### ISSUE TITLE
Trouble with Table Header and DataContainers

### ISSUE THREAD

**Issue by bnorm** [rjaros/kvision#246]
I'm having some trouble hooking up a Table and a DataConatiner and getting the header row to display properly. On it's own, I can create a Table without issue and the header displays as desired. However, when I use a table with a data container, the table header doesn't display and I'm not sure what I'm doing wrong.

```kotlin
class TestTab : SimplePanel() {
  init {
    text(value = "Table")
    val table = Table(
      types = setOf(TableType.SMALL, TableType.STRIPED, TableType.HOVER),
      headerNames = listOf("", "Name", "Type", "Actions"),
    )
    add(table)

    text(value = "DataContainer")
    dataContainer(
      model = mutableListOf<Resource>(),
      factory = { _, _, _ -> Row {} },
      container = Table(
        types = setOf(TableType.SMALL, TableType.STRIPED, TableType.HOVER),
        headerNames = listOf("", "Name", "Type", "Actions"),
      )
    )
  }
}
```

![image](https://user-images.githubusercontent.com/1074449/112732245-d3262400-8f06-11eb-8af5-3dc4459ee71d.png)

Could the issue be with the `disposeAll()` function in Table which DataConatiner calls when updating? This appears to clear the table header row as well and I didn't find anywhere it would be recreated.

**Comment by rjaros** [comment]
Hi and thanks for the feedback.
You are right, it looks like a bug in the `Table` component on the interaction with the `DataContainer`. It needs to be fixed.

**Comment by rjaros** [comment]
Fixed in 4.2.3

**Comment by bnorm** [comment]
Thanks! Confirmed it's fixed for me as well.
